### DISCUSSION TITLE
Is it possible to use authentification protocol SHA-224 to SHA-512 and privacy protocol AES-256 or AES-512 with snmp_exporter

### DISCUSSION THREAD

**Discussion by adjosan** [prometheus/snmp_exporter#1033] [Q&A] [answered]
Net-SNMP 5.8 and later support all the authentication protocols defined in [RFC7860|https://tools.ietf.org/html/rfc7860], which are:
- SHA-192
- SHA-256
- SHA-284
- SHA-512

Also starting with release 5.8 AES-192 and AES-256 are an optional configure option.

Is it possible to uses these protocols with snmp_exporter

**Comment by SuperQ** [comment, marked as answer]
Yes, this is is documented in the [generator README](https://github.com/prometheus/snmp_exporter/blob/f587fce813a83f22a0cda69ab6eb498da169bce8/generator/README.md). This has been supported since 2020.

**Reply by adjosan** [reply]
thanks :)
